File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var maxVowels = function(s, k) {
  const vowels = ["a", "e", "i", "o", "u"];
  let left = 0;
  let right = 0;
  let count = 0;
  let max = 0;
  
  for (let right = 0; right < s.length; right++) {
    if (vowels.includes(s[right])) {
      count++;
    }
    
    if (right > left + k - 1) {
      if (vowels.includes(s[left])) {
        count--;
      }
      
      left++;
    }
    
    if (count > max) {
      max = count;
    }
    
    if (count === k) {
      return k;
    }
  }
  
  return max;
};

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
      vowels = ["a", "e", "i", "o", "u"]
      left = 0
      right = 0
      maximum = 0
      count = 0

      for right in range(0, len(s)):
        if s[right] in vowels:
            count += 1

        if right - left + 1 > k:
          if s[left] in vowels:
            count -= 1

          left += 1

        maximum = max(maximum, count)

      return maximum
